kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/encryption/VaultTemplateInvocationContext.java

```diff
@@ -0,0 +1,148 @@
+import org.junit.jupiter.api.extension.TestTemplateInvocationContext;
+import org.junit.jupiter.api.extension.support.TypeBasedParameterResolver;
+import org.testcontainers.vault.VaultContainer;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.kroxylicious.filter.encryption.EnvelopeEncryption;
+import io.kroxylicious.filter.encryption.TemplateKekSelector;
+import io.kroxylicious.kms.provider.hashicorp.vault.VaultKms;
+import io.kroxylicious.kms.provider.hashicorp.vault.VaultKmsService;
+import io.kroxylicious.kms.provider.hashicorp.vault.VaultResponse;
+import io.kroxylicious.proxy.config.FilterDefinition;
+import io.kroxylicious.proxy.config.FilterDefinitionBuilder;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class VaultTemplateInvocationContext implements TestTemplateInvocationContext {
+    private static final String VAULT_TOKEN = "rootToken";
+    private static final String HASHICORP_VAULT = "hashicorp/vault:1.15";
+
+    private VaultContainer vaultContainer;
+    private VaultKms service;
+
+    @Override
+    public String getDisplayName(int invocationIndex) {
+        return VaultKms.class.getSimpleName();
+    }
+
+    @Override
+    public List<Extension> getAdditionalExtensions() {
+        return List.of(
+                new TypeBasedParameterResolver<TestContext>() {
+
+                    @Override
+                    public TestContext resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext)
+                            throws ParameterResolutionException {
+                        return new TestContext() {
+                            @Override
+                            public FilterDefinition getEncryptionFilterDefinition(String template) {
+                                return new FilterDefinitionBuilder(EnvelopeEncryption.class.getSimpleName())
+                                        .withConfig("kms", VaultKmsService.class.getSimpleName())
+                                        .withConfig("kmsConfig", Map.of("vaultUrl", vaultContainer.getHttpHostAddress(), "vaultToken", VAULT_TOKEN))
+                                        .withConfig("selector", TemplateKekSelector.class.getSimpleName())
+                                        .withConfig("selectorConfig", Map.of("template", template))
+                                        .build();
+                            }
+
+                            @Override
+                            public TestKekManager getTestKekManager() {
+                                return new TestKekManager() {
+                                    @Override
+                                    public CompletionStage<Void> generateKek(String alias) {
+                                        Objects.requireNonNull(alias);
+
+                                        try {
+                                            create(alias);
```

Comment:
technically we could inline the require non null here 
```suggestion
                                            create(Objects.requireNonNull(alias));
```

I'm on the fence if that improves readability or not.

Reply:
When I use Java `Objects.requireNotNull` I'm using them as you would use a precondition clause (in languages such as Eiffel or Ada where such things are first class).  Given those influences, it feels strange to bleed the preconditions (state invariant) into what the method actually does.

That all said, I'm happy to leave adopt a different practice if that's what we all want.

Reply:
I just find it strange to read a call I know returns and we "ignore" the return value. 

So I lean slightly towards using it inline, but happy to agree any convention and stick to it.

Reply:
@robobario do you have any preference?

Reply:
I prefer the guard clause approach, the throw at the start of the method is easier to read I find. Taking it further if you have multiple parameters to check I would much prefer that to all happen as a unit at the start of the method, rather than having requireNonNulls to read and parse scattered in the method.